packages/lesswrong/lib/collections/spotlights/spotlightScheduling.ts

```diff
@@ -0,0 +1,79 @@
+import sortBy from "lodash/sortBy";
+import maxBy from "lodash/maxBy";
+
+export const MS_IN_DAY = 1000 * 60 * 60 * 24;
+
+export interface SpotlightOrderingFields {
+  _id: string;
+  position: number;
+  lastPromotedAt: Date | string | number;
+  duration?: number | null;
+}
+
+export interface PromotionScheduleEntry<T extends SpotlightOrderingFields> {
+  spotlight: T;
+  activationDate: Date;
+  endDate: Date;
+}
+
+export const getDateValue = (value: Date | string | number | null | undefined) => {
```

Comment:
wat.  (if this is because `lastPromotedAt` is a nullable string which represents a date, we have `maybeDate` for that. I don't think `lastPromotedAt` should ever be a number.)